Apps/System/music.c

```diff
@@ -0,0 +1,461 @@
+    snprintf(progress_string, 6, "%d:%02d", s_progress / 60, s_progress % 60);
+
+    char length_string[6] = "";
+    snprintf(length_string, 6, "%d:%02d", s_length / 60, s_length % 60);
+
+    graphics_context_set_text_color(ctx, GColorBlack);
+    graphics_draw_text(ctx, s_artist,    fonts_get_system_font(FONT_KEY_GOTHIC_14),
+                       GRect(4, 108, 105, 10),                            GTextOverflowModeTrailingEllipsis,
+                       GTextAlignmentLeft, NULL);
+    graphics_draw_text(ctx, s_track,     fonts_get_system_font(FONT_KEY_GOTHIC_24_BOLD),
+                       GRect(4, 117, 110, 50),                            n_GTextOverflowModeWordWrap,//TODO remove 'n_' prefix
+                       GTextAlignmentLeft, NULL);
+    graphics_draw_text(ctx, progress_string,  fonts_get_system_font(FONT_KEY_GOTHIC_14),
+                       GRect( 4, 84, (144 - 30) / 2 + 4, 10),             GTextOverflowModeTrailingEllipsis, 
+                       GTextAlignmentLeft, NULL);
+    graphics_draw_text(ctx, length_string,    fonts_get_system_font(FONT_KEY_GOTHIC_14),
+                       GRect((144 - 30) / 2, 84, (144 - 30) / 2 - 4, 10), GTextOverflowModeTrailingEllipsis,
+                       GTextAlignmentRight, NULL);
+    graphics_draw_text(ctx, time_string, fonts_get_system_font(FONT_KEY_GOTHIC_14),
+                       GRect(0, 0, 113, 10),                              GTextOverflowModeTrailingEllipsis,
+                       GTextAlignmentCenter, NULL);
+    
+    if (s_animating_disk_change) {
+        draw_arm(ctx, s_arm_angle);
+        draw_record(ctx, s_next_disk_pos, s_next_disk_color);
+    }
+    draw_record(ctx, s_curr_disk_pos, s_curr_disk_color);
+
+    if(!s_animating_disk_change) {
+        draw_arm(ctx, s_arm_angle);
+    }
+    // Draw bar
+    graphics_fill_rect(ctx, GRect(4, 102, PROGRESS_PIXELS_MAX, 6), 1, n_GCornersAll);
+    graphics_context_set_fill_color(ctx, s_curr_disk_color);
+    graphics_fill_rect(ctx, GRect(5, 103, s_progress_pixels, 4), 0, GCornerNone);
+}
+
+static void music_window_load(Window *window) {
+    srand(50);
+    // srand(time(NULL)); TODO time seed
+    Layer *window_layer = window_get_root_layer(s_main_window);
+    GRect bounds = layer_get_unobstructed_bounds(window_layer);
+
+    s_curr_disk_color = GColorWindsorTan;
+    s_curr_disk_pos = RECORD_CENTER;
+    s_progress_pixels = 25;
+    s_trackcounter = 2;
+    get_new_track();
+    s_progress = 44;
+    s_arm_angle = ARM_HOME_ANGLE;
+    s_old_arm_angle = -1;
+    s_skip_value = 0;
+    s_main_layer = layer_create(bounds);
+    s_is_paused = false;
+    layer_add_child(window_layer, s_main_layer);
+
+    s_animation_record_ptr = animation_create();
+    animation_set_duration(s_animation_record_ptr, RECORD_SWOOSH_SPEED);
+    animation_set_implementation(s_animation_record_ptr, &implementation_record);
+    //animation_set_curve(s_animation_record_ptr, AnimationCurveEaseInOut);
```

Comment:
Yeah, I should implement this, probably with a lookup table or something.  :soon: